app/controllers/idv/phone_errors_controller.rb

```diff
@@ -0,0 +1,48 @@
+module Idv
+  class PhoneErrorsController < ApplicationController
+    include IdvSession
+
+    before_action :confirm_two_factor_authenticated
+    before_action :confirm_idv_needed
+    before_action :confirm_idv_session_started
+    before_action :confirmat_phone_confirmation_needed
+
+    def warning
+      @remaining_step_attempts = remaining_step_attempts
+    end
+
+    def timeout
+      @remaining_step_attempts = remaining_step_attempts
+    end
+
+    def jobfail
+      @remaining_step_attempts = remaining_step_attempts
+    end
+
+    def failure; end
+
+    private
+
+    def confirmat_phone_confirmation_needed
+      redirect_to_next_step if idv_session.user_phone_confirmation == true
+    end
+
+    def redirect_to_next_step
+      if phone_confirmation_required?
```

Comment:
Not covered

Reply:
Yep, I just pushed some tests for all of this